spreadsheet/row.go

```diff
@@ -121,6 +121,31 @@ func (r Row) Cells() []Cell {
 	return ret
 }
 
+// CellsWithEmpty returns a slice of cells including empty ones from the first column to the last one used in the sheet.
+// The cells can be manipulated, but appending to the slice will have no effect.
+func (r Row) CellsWithEmpty() []Cell {
+	maxColumnIdx := uint32(0)
+	for _, r := range r.s.SheetData.Row {
+		cells := r.C
+		if len(cells) > 0 {
+			lastCell := cells[len(cells) - 1]
+			ref, err := reference.ParseCellReference(*lastCell.RAttr)
+			if err != nil {
+				return []Cell{}
+			}
+			if maxColumnIdx < ref.ColumnIdx {
+				maxColumnIdx = ref.ColumnIdx
```

Comment:
would it make more sense to have a property on the sheet that returns MaxColumn() and then CellsWithEmpty(maxCol) as you had suggested?  Seems a bit wasteful to do this iteration every time

Reply:
Yes, it is good, though that means that cell should have an access to the Sheet struct (now it has access only to sml.Worksheet which is auto-generated and so cannot contain custom fields). I think it is reasonable that cell will know full info about the sheet it is located at.

Reply:
Would it make more sense if this function was defined on the sheet rather than the row?  I.e.
CellArea() which returns all rows and cells?  Seems a bit strange to do this on a row basis.

Reply:
you mean return all cells in the spreadsheet restricted by max used row and column?